src/v/storage/log.h

```diff
@@ -118,6 +120,8 @@ class log final {
      * segment: {[10,10][11,30][31,100][101,103]}
      * Truncate prefix at offset (31) will result in
      * segment: {[31,100][101,103]}
+     * Truncate prefix mid-batch (at offset 35) will result in
+     * segment {[31, 100][101,103]}, but reported start_offset will be 35.
```

Comment:
Not sure if this is the correct behavior, but this is what `disk_log_impl` does now.